R/selectors.R

```diff
@@ -65,18 +79,279 @@ selector__sub <- function(other) {
   }
 }
 
+selector__or <- function(other) {
+  if (is_polars_selector(other)) {
+    wrap_to_selector(
+      self$meta$`_as_selector`()$meta$`_selector_add`(other),
+      name = "or",
+      parameters = list(
+        self = self,
+        other = other
+      )
+    )
+  } else {
+    self$as_expr()$or(other)
+  }
+}
+
+selector__and <- function(other) {
+  if (is_polars_selector(other)) {
+    wrap_to_selector(
+      self$meta$`_as_selector`()$meta$`_selector_and`(other),
+      name = "and",
+      parameters = list(
+        self = self,
+        other = other
+      )
+    )
+  } else {
+    self$as_expr()$and(other)
+  }
+}
```

Comment:
Do we need `is_column` to match the Python implementation?
https://github.com/pola-rs/polars/blob/d51b12c706983b423e8ad344a5636ae903fe3431/py-polars/polars/selectors.py#L386-L400

Reply:
I added this for `&` and `|` but there's one part in the `&` code that I'm not sure about: https://github.com/pola-rs/polars/issues/19740

Reply:
Fixed upstream so I made the same change here